Sources/unitfrmpgndatabase.pas

```diff
@@ -0,0 +1,156 @@
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Chess 256.  If not, see <http://www.gnu.org/licenses/>.
+
+  Abstract:
+    This is a Chess PGN Database window.
+    This file Coded by Jim Kinsman <relipse@gmail.com>
+}
+unit unitFrmPgnDatabase;
+
+{$mode ObjFPC}{$H+}
+
+interface
+
+uses
+  Classes, SysUtils, Forms, Controls, Graphics, Dialogs, StdCtrls,
+  ExtCtrls, Grids, Menus, Types, BoardForms;
+
+type
+
+  { TfrmPGNDatabase }
+
+  TfrmPGNDatabase = class(TForm)
+    mniLoadGame: TMenuItem;
+    pnlDatabase: TPanel;
+    gridPGNDatabase: TStringGrid;
+    pmnPGNDatabase: TPopupMenu;
+    procedure gridPGNDatabaseDblClick(Sender: TObject);
+    procedure gridPGNDatabaseDrawCell(Sender: TObject; aCol, aRow: Integer;
+      aRect: TRect; aState: TGridDrawState);
+    procedure gridPGNDatabaseMouseMove(Sender: TObject; Shift: TShiftState; X,
+      Y: Integer);
+    procedure gridPGNDatabasePrepareCanvas(Sender: TObject; aCol,
+      aRow: Integer; aState: TGridDrawState);
+    procedure mniLoadGameClick(Sender: TObject);
+  private
+
+  public
+
+  end;
+
+var
+  frmPGNDatabase: TfrmPGNDatabase;
+  My_Grid_Highlight_BackGround_Color:Tcolor=ClGreen;
+  My_Grid_Highlight_Font_Color:Tcolor=clYellow;
+  My_Grid_Highlight_Font_Multiplier:Real=1.5;  // FOnt Multiplier for selected row
+  LastRow : Integer;
+implementation
+
+uses
+  MainUnit, GraphUtil, LCLIntf;
+{$R *.lfm}
+
+{ TfrmPGNDatabase }
+
+procedure TfrmPGNDatabase.gridPGNDatabaseDblClick(Sender: TObject);
+begin
+    mniLoadGameClick(nil);
+end;
+procedure TfrmPGNDatabase.gridPGNDatabaseDrawCell(Sender: TObject; aCol,
+  aRow: Integer; aRect: TRect; aState: TGridDrawState);
+begin
+  If Sender is TStringGrid then
+  begin
+    With (sender as TStringGrid) Do
+    Begin
+      Canvas.Brush.Style := bsSolid;
+      canvas.Font.Color:=clWhite;
+      if arow=row then       // Selected Row
+      begin
+        FocusColor:=My_Grid_Highlight_BackGround_Color;
+        Canvas.Brush.Color:=My_Grid_Highlight_BackGround_Color;
+        canvas.Font.Color:=My_Grid_Highlight_Font_Color;
+        canvas.Font.Size:=round(font.Size*My_Grid_Highlight_Font_Multiplier);
+      end
+      else
+      begin
+        canvas.Font.Color:=font.Color;
+        Canvas.Brush.Color:=color;
+        canvas.Font.Size:=font.Size;
+      end;
+      defaultdrawcell(Acol, Arow, arect, astate);
+    end;
+  end;
+end;
+
+procedure TfrmPGNDatabase.gridPGNDatabaseMouseMove(Sender: TObject;
+  Shift: TShiftState; X, Y: Integer);
+var
+  R, C:        Integer;
+  Start, Size: Integer;
+  i:           Integer;
+begin
+  gridPGNDatabase.MouseToCell(X, Y, c, r);
+  LastRow := r;
+end;
+
+procedure TfrmPGNDatabase.gridPGNDatabasePrepareCanvas(Sender: TObject; aCol,
+  aRow: Integer; aState: TGridDrawState);
+var
+  P: TPoint;
+  c, r: LongInt;
+  obe: Boolean;
```

Comment:
Use full and descriptive variable names where possible. `OldAllowOutBoundEvents`, for example.